log/sockstatlog/logger.go

```diff
@@ -50,10 +57,16 @@ type event struct {
 	Stats map[sockstats.Label]deltaStat `json:"s"`
 }
 
+// sockstatLogID reproducibly derives a new logid.PrivateID to be used for sockstat logging.
+// The returned PrivateID is the sha256 sum of id + "sockstat".
+func sockstatLogID(id string) logid.PrivateID {
```

Comment:
It's kinda gross that `id` is not type-safe, when I would expect this to be `logid.PublicID`. Unfortunately, the callers above this carry this variable around as a `string`. Perhaps, we should switch them all to specify `logid.PublicID`? Up to you whether to clean this up.

Reply:
Also, should this be exported so that we can call this logic from elsewhere?

Reply:
switched this to StableNodeID at Mihai's suggestion.  And yes, making it exported.  I'll come back and look at cleaning up the type safety on these other log IDs after.

Reply:
There is minor benefit of using public log ID over StableNodeID. The StableNodeID is broadcasted to other nodes in the tailnet, while the public log ID (to my recollection) is not as readily known by other nodes. This only changes the degree of spoofability (again, if we care).

Reply:
yeah, I suppose it's also a lot less disruptive to roll a new log ID if someone does start spoofing traffic than it would be to roll a new stable ID.  I think that's good enough reason to switch it back.